base/titer_model.py

```diff
@@ -149,6 +150,25 @@ def filter_strains(titers, strains):
         return {key: value for key, value in titers.iteritems()
                 if key[0] in strains and key[1][0] in strains}
 
+    @classmethod
+    def subset_to_date(cls, titers, node_lookup, date_range):
+        """Subset the given titers to the given date range based on the dates annotated
+        on each node.
+        """
+        # Filter titers from the future by keeping titers associated with
+        # strains from the given timepoint or earlier.
+        filtered_strains = set([
+            node_name
+            for node_name, node in node_lookup.items()
+            if node.numdate <= date_range[1] and node.numdate >= date_range[0]
+        ])
+        filtered_titers = cls.filter_strains(titers, filtered_strains)
+        original_counts = sum(cls.count_strains(titers).values())
+        filtered_counts = sum(cls.count_strains(filtered_titers).values())
+        sys.stderr.write("Filtered from %i to %i titers between %s and %s\n" % (original_counts, filtered_counts, date_range[0], date_range[1]))
+
+        return filtered_titers
```

Comment:
in this filtering, how do we deal with test vs reference viruses? What is the lower bound that is used here. references viruses are used for a long time...